e2e/common/common.steps.ts

```diff
@@ -75,6 +83,14 @@ class CommonSteps {
       .catch((e) => callback(e));
   }
 
+  @when(/clicks? on the "([^"]*)" link.*$/)
+  public clickOnLink(linkTitle: string, callback: CallbackStepDefinition): void {
+    this.world.app.clickLink(linkTitle)
+      .then(() => callback())
+      // it may fail but we still want to let tests continue
+      .catch((e) => callback(e));
+  }
```

Comment:
this note applies also to myself ;)

we may use either callback
```ts
  @when(/clicks? on the "([^"]*)" link.*$/)
  public clickOnLink(linkTitle: string, callback: CallbackStepDefinition): void {
    this.world.app.clickLink(linkTitle)
      .then(() => callback())
      // it may fail but we still want to let tests continue
      .catch((e) => callback(e));
  }
```
or return promise (instead of callback method attribute we define Promise as return value)
- specifyng both callback attribute and also promise return value throws error
- in this case data passed to `Promise.reject` (or `P.reject`) is displayed in test report as error message.
```ts
  @when(/^type "([^"]*)" into connection description/)
  public typeConnectionDescription(description: string): P<void> {
    // Write code here that turns the phrase above into concrete actions
    const connectionView = new ConnectionViewComponent();
    return connectionView.description.set(description);
  }
```